File: python/f40.loop.py
#!/usr/bin/python
import re
# spin loop as always true condition
#   break to exit from loop
#   continue to skip when there is no input
while True:
  # print a prompt and get answer
  answer = raw_input("Enter your name (quit to Exit): ")
  # skip loop if user enters an empty string
  if re.compile("^[\s\t]*$").match(answer):
    continue
  if answer == "quit":
    break                      # exit loop if user wants to quit
  print "Hello %s!" % answer   # output result as we are not exiting

File: windows/wsh.jscript/f40.loop.js
// spin loop as condition is always true
//  use break to exit out of loop
do {
  // prompt user
  WScript.stdout.write("Enter your name (quit to Exit): ");
  answer = WScript.stdin.readline();         // get input
  
  // skip loop if user enters nothing
  if (RegExp("^$").test(answer)) continue 
  
  // exit loop is user quits
  if (answer == "quit") break                                  
	  
  // output result as not exiting	  
  WScript.echo("Hello " + answer + "!"); 
} while (true)
